File: admincontrol/src/main/java/com/admincontrols/admincontrol/video/Video.java
package com.admincontrols.admincontrol.video;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "videos")
public class Video {
    @Id
    private String id;
    private String trainerId;
    private String workOutPlanId;
    private String title;
    private String description;
    private String imgUrl;
    private String videoUrl;
    private String modeOfInstruction;
    private String typeOfWorkout;
    private String caloriesBurnt;
    private int views;
    private List<String> tags;
    private List<String> likes;
    private List<String> dislikes;
    private boolean isApproved;
}

File: content-delivery/src/models/Video.js
const mongoose = require('mongoose');

const VideoSchema = new mongoose.Schema(
    {
        trainerId: {
            type: String,
            required: true
        },
        workOutPlanId: {
            type: String,
            required: true
        },
        title: {
            type: String,
            required: true,
        },
        description: {
            type: String,
            required: true,
        },
        imgUrl: {
            type: String,
        },
        videoUrl: {
            type: String,
            required: true
        },
        modeOfInstruction: {
            type: String,
            required: true
        },
        typeOfWorkout: {
            type: String,
            required: true
        },
        caloriesBurnt: {
            type: String,
            required: true
        },
        views: {
            type: Number,
            default: 0
        },
        tags: {
            type: [String],
            default: []
        },
        likes: {
            type: [String],
            default: []
        },
        dislikes: {
            type: [String],
            default: []
        },
        isApproved: {
            type: Boolean,
            default: false
        }
    },
    { timestamps: true }
);

module.exports = mongoose.model("Video", VideoSchema);